### DISCUSSION TITLE
Fabric setup problem with Ollama, localhost and ports problem

### DISCUSSION THREAD

**Discussion by ckmrt** [danielmiessler/Fabric#1047] [Q&A]
i installed fabric and ollama with open webui is running

then i tried to 

fabric --setup

this is what i receive

[Ollama]

Enter your Ollama URL (as a reminder, it is usually http://localhost:11434) (leave empty to skip):
http://localhost:11434
[Ollama] configured

because its localhost i tried 127.0.0.1:11434 as well
[OpenAI]

Enter your OpenAI API KEY (leave empty to skip):

[OpenAI] skipped
Ollama Get "http://localhost:3000/api/tags": dial tcp 127.0.0.1:3000: connect: connection refused

Available vendor models:

[Default]

Enter the index the name of your default model (leave empty to skip):

DEFAULT_MODEL=, is not valid

how can i solve this thank you

**Comment by Kltc55555** [comment]
Hi. Just installed from go (in WSL, Ubuntu 22.04).
I think i fixed this for me. So, step by step,  i ran these commands:

> -># cd ~/.config/fabric
> -># vi .env (you can use nano or any other editor you feel confortable with)
> 

Paste the following in the .env file or have it similar to this (i ran 'ollama run llama3.2' previously, but i'm guessing that you should be able to do this with any other model):

> PATTERNS_LOADER_GIT_REPO_URL=https://github.com/danielmiessler/fabric.git
> PATTERNS_LOADER_GIT_REPO_PATTERNS_FOLDER=patterns
> OLLAMA_API_URL=http://localhost:11434/
> DEFAULT_VENDOR=Ollama
> DEFAULT_MODEL=llama3.2:latest

After all this, you should be able to run 'fabric --setup' again, and configure what's missing from the previous point.
I'm guessing that, for some reason, the setup procedure isn't configuring ollama properly and just crashes.

I hope this helps you and anyone else who gets this.

**Comment by heavydluxe** [comment]
Yeah, something in the most recent Mac/ARM64 pull is converting 'localhost:11434' as an input in the --setup process to 'localhost/11434' in the environment file.  If you just view the .env file and put the corrected colon in place, everything runs from there.

Thank you @Kltc55555
